activerecord/lib/active_record/associations/association.rb

```diff
@@ -45,6 +45,8 @@ def initialize(owner, reflection)
 
         reset
         reset_scope
+
+        @no_strict_loading = nil
```

Comment:
From a naming perspective I would prefer `skip_strict_loading`. Otherwise this looks good to me 👍🏼